### PR TITLE
WIP: Testing possible Sarnia improvements.

### PR DESCRIPTION
**Pull request by sylus**
Just playing around with this myself and took your temp_fixes branch on top of 8.x-1.x and did the Search API Index load + creation as well as some more work on the Sarnia Entity + Backend.

Not sure how correct the additions are but can break this out into smaller commits or rebase this one ^_^

### CODE REVIEW

File: src/Plugin/search_api/backend/Sarnia.php [hunk 1]
```diff
@@ -147,37 +151,28 @@ public function getFieldNames(SearchApiIndex $index, $reset = FALSE) {
...
+    $solarium_query->addParam('qt', $conf['sarnia_request_handler']);
   }
 
   /**
    * Stash the results as loaded entities.
    */
-  protected function postQuery(array &$results, SearchApiQueryInterface $query, $response) {
+  protected function postQuery(ResultSetInterface $results, QueryInterface $query, $response) {
     parent::postQuery($results, $query, $response);
-    $sarnia_entity_type = sarnia_entity_server_name_load($this->server->machine_name);
```
Comment (sylus) [inline src/Plugin/search_api/backend/Sarnia.php]: How do we replace this logic?

File: src/Plugin/search_api/backend/Sarnia.php [hunk 2]
```diff
@@ -147,37 +151,28 @@ public function getFieldNames(SearchApiIndex $index, $reset = FALSE) {
...
-    entity_get_controller($sarnia_entity_type)->stash($results['results']);
```
Comment (sylus) [inline src/Plugin/search_api/backend/Sarnia.php]: was trying something like: 

`\Drupal::entityTypeManager->getStorage('sarnia')`

File: src/Form/SarniaTypeForm.php [hunk 3]
```diff
@@ -119,8 +120,21 @@ public function save(array $form, FormStateInterface $form_state) {
...
+        'id' => $type->get('id'),
+        'name' => $type->get('name'),
+        'description' => $type->get('name'),
+        'read_only' => 1,
+        'options' => [
+          'cron_limit' => 0,
+          'index_directly' => 0,
+        ],
+        'server' => $type->get('server_id'),
+        'datasources' => ['entity:sarnia'],
```
Comment (sylus) [inline src/Form/SarniaTypeForm.php]: Had no idea what is correct entry to put here.

**Comment by sylus** [comment]
Am doing battle with the _sarnia_entity_types function now and how best to replace it. I think the point was to not use any entity api so guess the following is to high a bootstrap?

<code>
function _sarnia_entity_types($reset = FALSE) {
  $types = &drupal_static(__FUNCTION__);
  if ((!isset($types) || $reset) && \Drupal::entityTypeManager()-> hasHandler('sarnia_type', 'storage')) {
    $storage = \Drupal::entityTypeManager()->getStorage('sarnia_type');
    $ids = $storage->getQuery()->execute();
    $types = $storage->loadMultiple($ids);
  }
  return $types;
}
</code>

**Comment by dcameron** [comment]
I'm really sorry, but I am totally unable to help out with this right now.  I wish I could devote some time for it, but I'm swamped at work.  I'll try to free up some personal time when I can.

**Comment by sylus** [comment]
No worries at all! You have already done a huge bulk of the work.

I am going to keep progressing as I have a need for this module soon for a site. If I reach a certain point of stability, I might get your stuff merged onto drupal.org and commit my addiitons on top of that, but that is still a few weeks off and I'll likely ping again before that :)

I'll keep this PR open so you can see my updates and if ever you see anything glaringly wrong let me know :)

**Pull request merged**